---
title: "Analysis"
author: "Mikhail Dozmorov"
date: "`r Sys.Date()`"
output:
  pdf_document:
    toc: no
  html_document:
    theme: cerulean
    toc: yes
---

```{r setup, echo=FALSE, message=FALSE, warning=FALSE}
# Set up the environment
library(knitr)
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', echo=F, dpi=100, warnings=F, message=F, comment=NA, warning=F, results='as.is', fig.width = 10, fig.height = 6) #out.width=700, 
library(pander)
panderOptions('table.split.table', Inf)
set.seed(1)
```

# Libraries

```{r libraries}
library(tidyverse)
library(readxl)
library(writexl)
library(GenomicRanges)
library(rtracklayer)
```

# Settings

```{r settings}
# Project folder path
dir_project <- "/Users/mdozmorov/Documents/Work/GitHub/CTCF.dev"

# Input files
# Predicted data
fileNameIn1 <- file.path(dir_project, "data/allcomp.txt.gz")
if (!file.exists(fileNameIn1)) {
  download.file("https://insulatordb.uthsc.edu/download/allcomp.txt.gz", fileNameIn1)
}
# Experimental data
fileNameIn2 <- file.path(dir_project, "data/CTCFBSDB_all_exp_sites_Sept12_2012.txt.gz")
# download.file("https://insulatordb.uthsc.edu/download/CTCFBSDB_all_exp_sites_Sept12_2012.txt.gz", fileNameIn2)
# Output files
fileNameOut1 <- file.path(dir_project, "RData", "hg18.CTCFBSDB.CTCF_predicted_human.RData")
fileNameOut1.1 <- file.path(dir_project, "RData", "hg18.CTCFBSDB.CTCF_predicted_human.bed")
fileNameOut2 <- file.path(dir_project, "RData", "mm8.CTCFBSDB.CTCF_predicted_mouse.RData")
fileNameOut2.1 <- file.path(dir_project, "RData", "mm8.CTCFBSDB.CTCF_predicted_mouse.bed")
# fileNameOut3 <- file.path(dir_project, "results/hg19.CTCFBSDB.CTCF_experimental_human.RData")
# fileNameOut3.1 <- file.path(dir_project, "results/hg19.CTCFBSDB.CTCF_experimental_human.bed")
# fileNameOut4 <- file.path(dir_project, "results/mm9.CTCFBSDB.CTCF_experimental_mouse.RData")
# fileNameOut4.1 <- file.path(dir_project, "results/mm9.CTCFBSDB.CTCF_experimental_mouse.bed")
```

# Load PREDICTED data

```{r data}
# Read in data
mtx <- read_tsv(fileNameIn1)
```

## EDA

```{r}
table(mtx$Species)
mtx_human <- mtx[mtx$Species == "Human", ]
mtx_mouse <- mtx[mtx$Species == "Mouse", ]
```

## GRanges

```{r}
mtx_to_gr <- function(mtx = mtx_human, genome_id = "hg19") {
  gr <- GRanges(seqnames = sapply(mtx$`Chromosome Location`, function(x) strsplit(x, ":|-")[[1]][1]), 
              IRanges(start = sapply(mtx$`Chromosome Location`, function(x) strsplit(x, ":|-")[[1]][2]) %>% as.numeric(), 
                      end = sapply(mtx$`Chromosome Location`, function(x) strsplit(x, ":|-")[[1]][3]) %>% as.numeric()))

  gr$`5PrimeGene` <- mtx$`5' Flanking Gene`
  gr$`3PrimeGene` <- mtx$`3' Flanking Gene`
  
  # Get chromosome info and match it to the chromosome order in ctcfBED
  chrom_data <- GenomeInfoDb::getChromInfoFromUCSC(genome = genome_id)
  # Subset to common chromosomes
  common_chromosomes <- intersect(chrom_data$chrom, seqlevels(gr))
  chrom_data <- chrom_data[chrom_data$chrom %in% common_chromosomes, ]
  gr <- keepSeqlevels(gr, common_chromosomes, pruning.mode = "tidy")
  # Match order
  chrom_data <- chrom_data[match(seqlevels(gr), chrom_data$chrom), ]
  # Check if chromosome order is the same
  if (!all.equal(seqlevels(gr), chrom_data$chrom)) {
    print(paste("Chromosome order does not match for", genome_id, "genome."))
    break
  }
  # Assign seqinfo data
  seqlengths(gr) <- chrom_data$size
  isCircular(gr) <- chrom_data$circular
  genome(gr)     <- genome_id

  return(gr)
}

hg18.CTCFBSDB.CTCF_predicted_human <- mtx_to_gr(mtx = mtx_human, genome_id = "hg18")
length(hg18.CTCFBSDB.CTCF_predicted_human)
summary(width(hg18.CTCFBSDB.CTCF_predicted_human))
mm8.CTCFBSDB.CTCF_predicted_mouse <- mtx_to_gr(mtx = mtx_mouse, genome_id = "mm8")
length(mm8.CTCFBSDB.CTCF_predicted_mouse)
summary(width(mm8.CTCFBSDB.CTCF_predicted_mouse))

# Save as RData object. subdir is the character name of the CTCF GRanges variable
save(list = "hg18.CTCFBSDB.CTCF_predicted_human", file = fileNameOut1)
save(list = "mm8.CTCFBSDB.CTCF_predicted_mouse", file = fileNameOut2)
# load(file = paste0(subdir, ".RData"))
# Save BED files
export.bed(hg18.CTCFBSDB.CTCF_predicted_human, fileNameOut1.1)
export.bed(mm8.CTCFBSDB.CTCF_predicted_mouse, fileNameOut2.1)
```

```{r eval=FALSE}
# Load EXPERIMENTAL data

# Read in data
mtx <- read_tsv(fileNameIn2)
```

```{r eval=FALSE}
## EDA

# head(mtx, n = 100) %>% View()
colnames(mtx)
table(mtx$Species)
table(mtx$Build)
mtx$Build[mtx$Build == "galgal3"] <- "galGal3"
table(mtx$ValidationStatus)
table(mtx$CellType)
table(mtx$Occupancy)
summary(mtx$CTCF_Rank %>% as.numeric())
summary(mtx$Readcount %>% as.numeric())
table(mtx$ENCODE)
summary(mtx$ENCODE_Peak)
table(mtx$M1M2_class)
```

```{r eval=FALSE}
## GRanges

mtx_to_gr <- function(mtx = mtx, org = "Rhesus") {
  mtx_selected <- mtx[mtx$Species == org, ]
  gr <- GRanges(seqnames = mtx_selected$Chr, IRanges(start = mtx_selected$GenomeBeginLoc, end = mtx_selected$GenomeEndLoc))

  gr$Reference        <- mtx_selected$Reference
  gr$ValidationStatus <- mtx_selected$ValidationStatus
  gr$UpstreamGene     <- mtx_selected$UpstreamGene
  gr$DownstreamGene   <- mtx_selected$DownstreamGene
  gr$CellType         <- mtx_selected$CellType
  gr$Occupancy        <- mtx_selected$Occupancy
  gr$CTCF_Rank        <- mtx_selected$CTCF_Rank
  gr$Readcount        <- mtx_selected$Readcount
  gr$ENCODE           <- mtx_selected$ENCODE
  gr$ENCODE_Peak      <- mtx_selected$ENCODE_Peak
  gr$M1M2_class       <- mtx_selected$M1M2_class
  
  genome_id <- unique(mtx_selected$Build)
  # Get chromosome info and match it to the chromosome order in ctcfBED
  chrom_data <- GenomeInfoDb::getChromInfoFromUCSC(genome = genome_id)
  # Subset to common chromosomes
  common_chromosomes <- intersect(chrom_data$chrom, seqlevels(gr))
  chrom_data <- chrom_data[chrom_data$chrom %in% common_chromosomes, ]
  gr <- keepSeqlevels(gr, common_chromosomes, pruning.mode = "tidy")
  # Match order
  chrom_data <- chrom_data[match(seqlevels(gr), chrom_data$chrom), ]
  # Check if chromosome order is the same
  if (!all.equal(seqlevels(gr), chrom_data$chrom)) {
    print(paste("Chromosome order does not match for", genome_id, "genome."))
    break
  }
  # Assign seqinfo data
  seqlengths(gr) <- chrom_data$size
  isCircular(gr) <- chrom_data$circular
  genome(gr)     <- genome_id

  return(gr)
}

organisms <- unique(mtx$Species)
# organisms <- organisms[organisms != c("Rhesus")]
# for(i in organisms) {
#   print(i)
#   gr <- mtx_to_gr(mtx, org = i)
#   # Output files
#   fileNameOut1 <- file.path(dir_project, paste0("results/CTCF_", i, "_CTCFBSDB_experimental.RData"))
#   # Save as RData object. 
#   # save(list = mtx_to_gr(mtx = mtx_human, genome_id = "hg19"), file = fileNameOut1)
# }

# load(file = paste0(subdir, ".RData"))

hg18.CTCFBSDB.CTCF_experimental_human <- mtx_to_gr(mtx = mtx, org = "Human")
length(hg18.CTCFBSDB.CTCF_experimental_human)
summary(width(hg18.CTCFBSDB.CTCF_experimental_human))
```


