- http://cisbp.ccbr.utoronto.ca/TFreport.php?searchTF=T094831_2.00 - `CisBP_2022_02_24_4_13_pm` - Homo Sapiens CTCF PWM motif matrices


 mESC CTCF, https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE36027
- mESC epigenomic marks, https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE31039

- RenLab Hi-C and CTCF data http://chromosome.sdsc.edu/mouse/download.html

- Find all available CTCF datasets at http://cistrome.org/db/#/. Need to select relevant tissue-specific marks.

 Motif finder-identified CTCF sites: Potential CTCF motifs across provided genomes are available at http://hicfiles.s3.amazonaws.com/internal/motifs/GENOME_ID.motifs.txt (e.g. http://hicfiles.s3.amazonaws.com/internal/motifs/hg19.motifs.txt). hg19, hg38, mm9, and mm10 supported

## Other transcription factors

- CEBPB, CMYC, CTCF, JUND, MAFK, P300, POL2, POLR2A, RAD21, SMC3, TAF1, and TBP for hESCs, and CEBPB, CTCF, MAFK, POLR2A, and RAD21 - From Arboretum-Hi-C paper

- In addition to CTCF, ZNF143, YY1, DNAse, H3K36me3, TSSs, RNA Pol II, SP1, ZNF274, SIX5. From Hong, Seungpyo, and Dongsup Kim. “Computational Characterization of Chromatin Domain Boundary-Associated Genomic Elements.” Nucleic Acids Research 45, no. 18 (October 13, 2017): 10403–14. https://doi.org/10.1093/nar/gkx738.

- four members of the cohesin complex (STAG2, SMC3, SMC1A, and RAD21) - from M. Ryan Corces and Victor G. Corces, “The Three-Dimensional Cancer Genome,” Current Opinion in Genetics & Development 36 (2016)

# Motif matrices

https://insulatordb.uthsc.edu/download/CTCFBSDB_PWM.mat

# Backburner

CTCF-MP, https://github.com/ma-compbio/CTCF-MP/tree/master/Data/gm12878


