---
# csl: /Volumes/GoogleDrive/My Drive/manuscript_template/styles.ref/cell-reports.csl
csl: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/styles.ref/cell-reports.csl
output: 
  word_document:
#    reference_docx: /Volumes/GoogleDrive/My Drive/manuscript_template/styles.doc/NIH_grant_style.docx
    reference_docx: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/styles.doc/NIH_grant_style.docx
# https://github.com/mdozmorov/manuscript_template/tree/master/references
# bibliography: /Volumes/GoogleDrive/My Drive/manuscript_template/references/references_MD.bib
bibliography: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/references/references_MD.bib
always_allow_html: true
---

```{r echo=FALSE, message=FALSE, warning=FALSE}
# Set up the environment
library(knitr)
opts_chunk$set(cache.path='cache/', fig.path='img/', cache=F, tidy=T, fig.keep='high', echo=F, dpi=100, warnings=F, message=F, comment=NA, warning=F, results='as.is', fig.width = 10, fig.height = 6) #out.width=700, 
library(pander)
panderOptions('table.split.table', Inf)
set.seed(1)
# library(kableExtra)
library(readxl)
dir_data <- "/Users/mdozmorov/Documents/Work/GitHub/CTCF.dev"
```

# CTCF: A Bioconductor data package providing genomic coordinates of human and mouse CTCF binding sites

# Authors (alphabetical order)

Mikhail Dozmorov *  
Wancen Mu  
Eric Davis  
Stuart Lee  
Tim Triche  
Douglas Phanstiel  
Michael Love  

# Abstract

CTCF (CCCTC-binding factor) is considered the master regulator of the genome's structure and function. Numerous studies investigated the anatomy of CTCF binding, revealing the diversity of CTCF motifs. This diversity leads to the lack of a unified source of CTCF binding data. We collected position weight matrices of CTCF binding motifs and defined strand-oriented CTCF binding sites in human and mouse genome assemblies, including the Telomere to Telomere human genome. We also included selected experimentally determined and predicted CTCF binding sites, such as CTCF-bound cis-regulatory element from SCREEN ENCODE. We recommend filtering less significant CTCF binding motifs and demonstrate that liftOver is a viable option to convert CTCF coordinates between genome assemblies. Our data is assembled in the CTCF Bioconductor package (https://bioconductor.org/packages/CTCF/) and is available on AnnotationHub.

### Keywords

# Introduction

The structural and regulatory organization of the mammalian genome is fundamentally dependent on CTCF (CCCTC-binding factor), a versatile transcription regulator evolutionary conserved from fruit fly to human [@Phillips:2009aa; @Ohlsson:2001wu; @Kim:2007ux]. Originally defined as a repressor/insulator protein [@Lobanenkov:1990vm; @Filippova:1996tv], it has been found to be involved in a variety of regulatory functions including transcriptional activation, imprinting, X-chromosome activation, intra- and inter-chromosomal contacts in three dimensions [@Phillips:2009aa; @Ohlsson:2010uy]. CTCF is ubiquitously expressed, and its deletion in the germline cells is incompatible with cell viability [@Heath:2008vs]. Mutations in CTCF binding sites and CTCF itself are associated with human cancers [@Liu:2019vg; @Saldana-Meyer:2019aa], identifying CTCF as an important candidate tumor-suppressor gene [@Filippova:2002uz].

<!-- Most CTCF binding sites are located far from the transcriptional start sites, with their distribution strongly correlated with genes [@Kim:2007ux]. binding motif length varying between 20 and 40bp -->

CTCF binds to different DNA sequences through combinatorial use of its 11-zinc finger domains to target sites that have remarkable sequence variation [@Filippova:1996tv; @Ohlsson:2001wu; @Jolma:2013vd]. A 15-20bp CTCF consensus motif from the ChIP-seq data analysis has been defined [@Kim:2007ux], referred to hereafter as M1. This motif interacts with the N-terminal zinc fingers for most CTCFâ€“DNA-binding events. Subsequently, a shorter 9bp motif (M2) interacting with the C-terminal fingers was discovered that, together with M1, forms CTCF binding sites spanning 34-35bp [@Schmidt:2012uk]. The complexity of CTCF binding was also noted in the earlier study of motif discovery in conserved noncoding elements by Xie et al. [@Xie:2007wm], detecting LM3, LM7, and LM23 motifs as parts of CTCF binding. Subsequently, $\sim80\%$ of CTCF conserved motifs were found to be bound by ZFs 4-7, while ZFs 1-2 and 8-11 bind to nonconserved flanking sequences, and ZFs 9-11 were associated with a second conserved motif [@Nakahashi:2013uw; @Pugacheva:2020wz]. Studies have reported 20,000-50,000 CTCF binding sites within the human and mouse genomes [@Kim:2007ux; @Nakahashi:2013uw; @Wang:2012ty].

<!-- CTCF binding has long been known to be critical for Topologically Associating Domain (TAD) and chromatin loop formation [@Dixon:2012aa; @Oti:2016aa; @Vietri-Rudan:2015aa]. Loops are formed via cohesin-mediated loop extrusion mechanism [@Alipour:2012aa; @Fudenberg:2016aa; @Goloborodko:2016vz; @Schwarzer:2016aa] with CTCF (in convergent orientation) and cohesin marking domain boundaries [@Nishana:2020tv; @Rao:2014aa; @Vietri-Rudan:2015tu; @Beagan:2020wz; @Stilianoudakis:2020aa; @Pugacheva:2020wz]. To demonstrate the use of CTCF data package, we provide a Python script that quantifies the proportion of loops -->

CTCF binding profiles between different tissues exhibit substantial concordance [@Kim:2007ux]. For instance, up to 70% of binding sites are common between any two given cell types, while variability appears to result from differential DNA methylation [@Wang:2012ty]. We aimed to provide uniformly detected CTCF binding sites for human and mouse genomes, the top two species by the number of experiments in GEO. Given the differences in genome assemblies, we defined CTCF binding sites directly for each assembly, including the recently created Telomere to Telomere human genome assembly (T2T) [@Nurk:2022up]. We demonstrate that coordinate conversion of CTCF binding sites between genome assemblies (liftOver) is a viable way to obtain assembly-specific CTCF binding sites. We further demonstrate the need for filtering less significant CTCF binding sites. Given the importance of CTCF binding in convergent orientation for forming chromatin loops [@Rao:2014aa; @Nanni:2020aa; @Vietri-Rudan:2015tu], we provide strand-specific CTCF binding sites indicating the directionality of CTCF binding. We also include selected experimental CTCF binding sites to provide a comprehensive general-purpose collection of CTCF binding profiles. The CTCF R package assembles human and mouse CTCF binding data accessible in the R/Bioconductor ecosystem via AnnotationHub. 

# Results

CTCF binding specificities in the form of Position Weight Matrices (PWMs) [@Stormo:2000uw] are available from multiple databases. The JASPAR 2022 [@Castro-Mondragon:2022uv] database lists one 19bp-long CTCF PWM (MA0139.1) and two 34 and 35bp-long PWMs (`Table 1`). The HOCOMOCO v11 [@Kulakovskiy:2018vw] database lists one CTCF PWM (H11MO.0.A) that is highly similar to the 19bp PWM in JASPAR (`Figure 1A`). Although human and mouse CTCF motifs are considered highly conserved [@Xie:2007wm], the HOCOMOCO database provides a mouse-specific CTCF PWM that is expectedly similar to human. SwissRegulon [@Pachkov:2013uk] provides one 20bp-long CTCF PWM similar to the JASPAR and HOCOMOCO PWMs. The 34bp and 35bp JASPAR PWMs formed their own cluster, together with the CTCF PWM identified in Jolma et al. study [@Jolma:2013vd]. The Jolma et al. study analyzed the sequence-specific binding of human TFs using high-throughput SELEX and ChIP-sequencing and described 239 distinctly different binding specificities, including a 17bp long CTCF motif dubbed CTCF_full [@Jolma:2013vd]. The CTCF binding site database (CTCFBSDB) provides multi-organism computationally predicted and experimentally determined from the literature CTCF binding sites [@Ziebarth:2013vj] and includes PWMs for the M1+M2 motifs identified by Schmidt et al. [@Schmidt:2012uk], the Ren_20 motif identified by Kim et al. [@Kim:2007ux], and the LM2, LM7, and LM23 motifs identified by Xie et al. [@Xie:2007wm] (`Figure 1B`). The CIS-BP database [@Weirauch:2014ul] lists 83 directly determined human-specific binding motifs (8 require Transfac license) ranging from 11 to 21bp (median - 15bp), and two mouse-specific motifs (`Table 1`, `Supplementary Figure 1`). The number of human-specific CTCF PWMs likely reflects the variability of CTCF binding sequences without considering the M1+M2 combinatorial structure. These PWMs were used to scan the human and mouse genomes for CTCF binding motifs.

```{r}
fileNameIn1 <- file.path(dir_data, "Figures", "Figure_1.png")
knitr::include_graphics(fileNameIn1)
```
**Figure 1. CTCF motif logos of PWMs used to obtain CTCF binding sites.** PWMs from (A) MEME and (B) CTCFBSDB databases. Motifs were clustered using the `motifStack` R package.

**Table 1. CTCF PWM information.** "Motif" - individual motif IDs or the total number of motifs per database; "Length" - motif length of the range of lengths; "URL" - direct links to motif pages; "Ref" - literature reference.
```{r eval=TRUE}
fileNameIn1 <- file.path(dir_data, "Tables", "Table_1_PWMs.xlsx")
mtx <- read_xlsx(fileNameIn1)
mtx[is.na(mtx)] <- ""
kable(mtx)
```

We used the FIMO tool from the MEME suite [@Bailey:2009aa] as one of the oldest and most well-known tool for motif scanning. The tool uses a dynamic programming algorithm to convert log-odds scores of motif matching into p-values, assuming a zero-order background model [@Cuellar-Partida:2012wj]. The default p-value cutoff used to generate the package's data is 1E-4. However, we observed that up to 98% of motifs have a detection p-value more than 1E-6 (`Figure 2A`, `Supplementary Figure 2A`), suggesting most CTCF motifs have a relatively poor match. Filtering results at 1E-6 allows focusing on more reliably detected CTCF sites. Furthermore, given some databases provide multiple CTCF PWMs, FIMO detects many overlapping CTCF sites (`Supplementary Table S1`). Reducing them (merging overlapping CTCF sites) in combination with 1E-6 cutoff filtering yields the number of CTCF sites comparable to previously reported 20,000-50,000 sites. Although we provide FIMO results as-is, we recommend filtering and reducing them.

```{r}
fileNameIn1 <- file.path(dir_data, "Figures", "Figure_2.png")
knitr::include_graphics(fileNameIn1)
```
**Figure 2. Properties of CTCF motifs detected by FIMO.** (A) Density plot of p-value distributions of motifs detected using different motif databases in the hg38 genome assembly. Red dashed line - 1E-6 p-value cutoff. (B) Jaccard overlaps among CTCF binding sites detected in the original and liftOver genome assemblies. CTCF sites were detected using JASPAR 2022 MA0139.1 PWM. The correlogram was clustered using Euclidean distance and Ward.D clustering method. White-red gradient indicate low-to-high Jaccard overlaps. Jaccard values are shown in the corresponding cells.

**Supplementary Table S1. Summary of CTCF binding data provided in the package.** CTCF sites for each genome assembly and PWM combination were detected using FIMO. "ID" - object names formatted as `<assembly>.<database name>`; "Assembly" - genome assembly, T2T - telomere to telomere (GCA_009914755.4) genome assembly; "All (p-value threshold Xe-Y)" - the total number of CTCF binding sites in the corresponding BED file at the 1E-4 or 1E-6 threshold; "Non-overlapping (p-value 1E-4 or 1E-6 threshold)" - number of non-overlapping CTCF binding sites (overlapping regions are merged) at the 1E-4 or 1E-6 threshold.

Several databases and papers provide predicted and experimental CTCF binding information. The ENCODE Phase III data release has created SCREEN, a registry of cis-regulatory element (cCRE) regions for human (hg38) and mouse (mm10) genome assemblies [@ENCODE-Project-Consortium:2020uq] (`Table 2`). The human set of CTCF-bound cCREs contains 450,641 records (median size 260bp), classified into distal-bound Enhancer-Like Signature (dELS) regions (61.18%), proximal ELS (pELS) regions (21.48%), CTCF-only sites (7.95%), promoter-like signature (PLS) regions (6.98%), and DNase-H3K4me3 regions bound by CTCF (1.84%). The mouse set contained 82,777 records (median size 284bp) similarly classified into 36.46% dELS-bound CTCF sites, 25.62% CTCF-only sites, 20.74% pELS-bound, 13.06 PLS-bound, and 4.11% DNase-H3K4me3-bound CTCF sites. A limitation of this data is that it lacks strand information limiting the opportunity to use this data for CTCF orientation-specific analysis. We added human and mouse CTCF-bound cCREs (hg38 and mm10 genome assemblies) as experimentally validated CTCF binding sites.

**Table 2. Predefined CTCF binding data.** "Database" - source of data; "Number" - number of binding sites; "Assembly" - genome assembly; "URL" - direct link to data download; "Ref" - literature reference. 

The CTCFBSDB database [@Bao:2008vq] assembled predicted CTCF binding information for human (13,401 20bp-long sites) and mouse (5,504 sites) genomes. The database has not been updated since 2013 and the predictions are available for hg18 and mm8 genome assemblies only (`Table 2`). As coordinate conversion between genome assemblies (liftOver) is a well-established practice, we investigated Jaccard overlap among CTCF sites aligned to hg18, hg19, hg38, and T2T genome assemblies and their liftover counterparts. Expectedly, genome assembly was the primary driver of similarity among sets of CTCF sites, irrespectively whether they were detected in the original genome assembly or lifted over from those detected in other assemblies (`Figure 2B`). liftOver among hg18, hg19, and hg38 showed the highest Jaccard overlap (0.96-1.00), while CTCF sites lifted from hg19 and hg38 to T2T showed less overlap (0.92-0.93) due to the relatively large number of unmapped sites (`Supplementary Table S2`). The mm10/mm9-detected CTCF binding sites also demonstrated good overlap between the originally detected and liftOver versions (`Supplementary Figure 2B`).  These results indicate that liftOver is a viable alternative to obtain CTCF sites in the required genome assembly. Using liftOver, we provide hg19/hg38 and mm9/mm10 versions of predicted CTCF sites from the CTCFBSDB database.

**Supplementary Table S2. liftOver statistics.** "File name" - file names prefixed with genome assembly. Files with sites directly detected from a genome assembly have suffix ".MA0139.1.FIMO". Sites that can't be mapped are outputted into the "Unmapped." files. "Number of CTCF sites" - CTCF binding site counts.


<!-- Chang et al. provide CTCF ChIP-seq peaks from mouse ESCs [@Chang:2021wl]. They report 174,268 peaks (83,522 unique regions) having at least one significant CTCF binding motif (mm10 genome assembly). It should be noted that mouse embryonic stem cells (ESCs) have thousands of CTCF-binding events that cannot be conserved in the human genome, as they are found in rodent-specific B2 repeat elements [@Bourque:2008ws]. -->

# Discussion

Our goal was to provide easy programmatic access to CTCF binding data applicable to any cell or tissue type. However, despite the high conservation of CTCF binding across cell types, cell-type-specific CTCF binding may be of interest. Many databases, such as ENCODE, GEO, UniBind, ReMap, CistromeDB, and individual publications [@Oti:2016wq; @Chang:2021wl; @Keilwagen:2019uv] provide download and/or REST API access to cell-type-specific data.<!--Recent methods for cell-type-specific prediction of TFBSs, such as Catchitt, also provide downloadable data.--> Within the R ecosystem, Bioconductor's AnnotationHub cell type-specific human and mouse CTCF binding data from ENCODE. Given the use of cell-type-specific data is specialized and the data are frequently genome assembly-specific, we did not include cell-type-specific data in our package.

Besides FIMO, other tools for motif scanning can be utilized. They have their own algorithms and defaults which may lead to different results. The popular HOMER motif screening tool [@Heinz:2010aa] has several options, including its unique autonormalization of the sequencing bias algorithm. The PWMscan genomewide motif scanning web tool [@Ambrosini:2018wv] uses its own matrix_scan program. Furthermore, PWMscan has the default background nucleotide composition A=0.29, C=0.21, G=0.21, T=0.29 instead of the FIMO's default A=0.25, C=0.25, G=0.25, T=0.25, and uses a 1E-5 p-value threshold instead of 1E-4 for FIMO. PWMscan benchmarked its tool and showed over 99% agreement with FIMO, HOMER, and other motif scanning tools. Our data includes results using the FIMO tool and we expect these results to be generally consistent when using other tools.

<!-- The ENCODE project provides CTCF motif models derived from human ChIP-seq data using curated motif models from literature (JASPAR MA0139.1 and jolma_full_M2 PWMs) and de novo motif models discovered from five motif discovery tools (10 PWMs) [@Kheradpour:2014we] (http://compbio.mit.edu/encode-motifs/. We did not use these PWMs because of differences in tool-specific format, cell-type-specificity of de novo PWMs, and inconsistencies of curated motif models to current databases. -->

The gene CTCF was duplicated in early amniotes, giving rise to a paralogue called "Brother of Regulator of Imprinted Sites" (BORIS, or CTCFL) [@Loukinov:2002vc; @Hore:2008wr]. CTCFL became progressively specialized to testis-specific expression throughout amniote evolution and is considered to normally be expressed only in germ cells.<!--but also aberrantly activated in numerous cancers [@Pugacheva:2010tl; @Nishana:2020tv]--> Human CTCF and CTCFL have on average 99.5% identity within the ZF domain but much less conserved in N and C terminal domains. CTCFL and CTCF binding partly overlap, but given that its binding does not overlap with cohesin [@Sleutels:2012vd], it cannot insulate chromosome domains [@Nishana:2020tv]. Because of CTCFL's tissue-specific expression and distinct functionality, we did not include its binding information. <!--CTCFL sites had less of a requirement for the "A" base in the triplet where ZF7 binds and an increase for "C" in the triplet that was bound by ZF4 [@Nishana:2020tv; @Sleutels:2012vd].-->

# Methods

MEME motif databases (updated 20 Mar 2022) containing HOCOMOCO, CIS-BP, JASPAR2022, Jolma 2013, and SwissRegulon motifs in MEME format were downloaded from https://meme-suite.org/meme/meme-software/Databases/motifs/motif_databases.12.23.tgz. Motifs from the CTCFBSDB database were downloaded from https://insulatordb.uthsc.edu/download/CTCFBSDB_PWM.mat and converted to MEME format using `transfac2meme -use_acc CTCFBSDB_PWM.mat > CTCFBSDB_PWM.meme`. Human and mouse motifs from CIS-BP were downloaded via the bulk download interface, combined and converted to the MEME format using a custom script. Motifs were visualized and clustered using the `motifStack` v.1.38.0 R package. Jolma 2013 and CTCFBSDB EMBL_M1 motifs were converted to their reverse complements to match the orientation of other motifs (`universalmotif` v.1.24.4). 

CTCF binding sites were detected using the FIMO tool from the MEME v.4.11.2 suite [@Bailey:2009aa] using default parameters and the `--max-stored-scores 1000000` setting to avoid the automatic drop of less significant motifs in case of memory limits. Softmasked genome sequences were downloaded from the USCS Genome Browser database except for the T2T genome (chm13v2.0) from the https://github.com/marbl/CHM13 repository. Motifs were scanned separately for each chromosome (autosomal and sex chromosomes only), and the results were combined. We used human- and mouse-specific matrices for the corresponding genomes (HOCOMOCO, CIS-BP). As JASPAR, CTCFBSDB, and CIS-BP databases provide several CTCF motifs, all database-specific motifs were used to scan the genomes; consequently, the detected CTCF sites may overlap due to motif similarity. We provide data as GenomicRanges objects and BED files with "seqnames", "start", "end", "name", "score", "strand", "width", "pvalue", "qvalue", "sequence" columns.

To investigate the effect of coordinate conversion, we used `liftOver` v.377 on CTCF binding sites (JASPAR 2022 MA0139.1) aligned to hg18, hg19, hg38, and T2T (NCBI GCA_009914755.4 assembly) genomes. The conversion chains were obtained from the UCSC Genome Browser database except for the T2T genome providing chains at the https://github.com/marbl/CHM13 repository. We performed all conversions except hg38->hg18 and hg18->T2T, T2T->hg18 due to the lack of the corresponding chains. Jaccard overlap matrix between the original and liftOver CTCF binding sites was clustered using Euclidean distance and ward.D clustering method and visualized using the `pheatmap` v.1.0.12 R package.

Predefined CTCF binding sites were obtained from the SCREEN ENCODE and CTCFBSDB databases. The hg18 and mm8 CTCF binding sites were converted to hg19/hg38 and mm9/mm10 versions using `liftOver`. We did not include CTCFBSDB data compiled from public experimental data because 1) mouse data was lacking, 2) hg18 genome assembly limits usability, and 3) region size was highly variable, ranging from 2bp to >350Kb.

# Abbreviations

# Acknowledgements

_Conflict of Interest._ None.

# Funding

This work was supported in part by the George and Lavinia Blick Research Fund scholarship to MD.

# Tables

# Figure legends

# Supplementary material

# References
