---
# csl: /Volumes/GoogleDrive/My Drive/manuscript_template/styles.ref/cell-reports.csl
csl: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/styles.ref/cell-reports.csl
output: 
  word_document:
#    reference_docx: /Volumes/GoogleDrive/My Drive/manuscript_template/styles.doc/NIH_grant_style.docx
    reference_docx: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/styles.doc/NIH_grant_style.docx
# https://github.com/mdozmorov/manuscript_template/tree/master/references
# bibliography: /Volumes/GoogleDrive/My Drive/manuscript_template/references/references_MD.bib
bibliography: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/references/references_MD.bib
---

Authors

# Abstract

### Keywords

# Introduction

The structural and regulatory organization of the mammalian genome is fundamentally dependent on CTCF (CCCTC-binding factor), a versatile transctiprion regulator that is evolutionary conserved from fruit fly to human [@Phillips:2009aa][@Ohlsson:2001wu][@Kim:2007ux]. Originally defined as a repressor/insulator protein [@Lobanenkov:1990vm][@Filippova:1996tv], it has been found to be involved in a variety of regulatory functions including transcriptional activation, imprinting, X-chromosome activation, the intra- and inter-chromosomal contacts in three dimensions [@Phillips:2009aa][@Ohlsson:2010uy]. Point mutation and loss of heterozygosity of CTCF is associated with human cancer, identifying CTCF as an important candidate tumor-suppressor gene [@Filippova:2002uz].

Most of CTCF binding sites are located far from the transcriptional start sites, with their distribution strongly correlated with genes [@Kim:2007ux].

CTCF localization is largely invariant across different cell types [@Kim:2007ux]. The goal of our package is to provide such lists of genome-wide CTCF binding sites generally applicable across cell types.


CTCF binds to different DNA sequences through combinatorial use of 11-zinc finger domains to target sites that have remarkable sequence variation with binding motif length varying between 20 and 40bp [@Filippova:1996tv][@Ohlsson:2001wu]. A 15-20bp CTCF consensus motif from the analysis of ChIP-seq data has been defined [@Kim:2007ux], referred hereafter as M1. It was subsequently refined to include an additional 9bp motif (M2) [@Schmidt:2012uk]. For the majority of CTCFâ€“DNA- binding events, the N-terminal zinc fingers interact with a 14-bp long M1 motif. Additionally, the C-terminal fingers interact with a shorter M2 sequence, creating a 34-bp-long M1+M2 motif. The complexity of CTCF binding was also noted in the earlier study of motif discovery in conserved noncoding elements by Xie et al. [@Xie:2007wm], detecting LM3, LM7, LM23 motifs as parts of CTCF binding. Studies have reported 20,000-50,000 CTCF binding sites within the human genome [@Kim:2007ux][@Nakahashi:2013uw].

Position Weight Matrices (PWMs) [@Stormo:2000uw] for CTCF are available from multiple databases. JASPAR 2022 [@Castro-Mondragon:2022uv] and HOCOMOCO [@Kulakovskiy:2018vw] databases list one 19bp-long CTCF PWM ([MA0139.1](https://jaspar.genereg.net/matrix/MA0139.1/) and [CTCF_HUMAN.H11MO.0.A](https://hocomoco11.autosome.org/motif/CTCF_HUMAN.H11MO.0.A), respectively). Cis-BP database [@Weirauch:2014ul] lists 83 directly determined human-specific [CTCF](http://cisbp.ccbr.utoronto.ca/TFreport.php?searchTF=T094831_2.00) binding motifs, likely reflecting the variability of CTCF motifs without considering the M1+M2 combinatorial structure. The CTCF binding site database (CTCFBSDB) provides multi-organism computationally predicted and experimentally determined from the literature CTCF binding sites [@Ziebarth:2013vj] and includes PWMs for the M1+M2 motifs identified by Schmidt et al. [@Schmidt:2012uk], the Ren_20 motif identified by Kim et al. [@Kim:2007ux], and the LM2, LM7, and LM23 motifs identified by Xie et al. [@Xie:2007wm]. 

Unfortunately, the database has not been updated since 2013 and its data are limited to old genome assemblies (e.g., hg19, mm9).

CTCF binding has long been known to be critical for Topologically Associating Domain (TAD) and chromatin loop formation [@Dixon:2012aa][@Oti:2016aa][@Vietri-Rudan:2015aa]. Loops are formed via cohesin-mediated loop extrusion mechanism [@Alipour:2012aa][@Fudenberg:2016aa][@Goloborodko:2016vz][@Schwarzer:2016aa] with CTCF (in convergent orientation) and cohesin marking domain boundaries [@Nishana:2020tv][@Rao:2014aa][@Vietri-Rudan:2015tu][@Beagan:2020wz][@Stilianoudakis:2020aa]. To demonstrate the use of CTCF data package, we provide a Python script that quantifies the proportion of loops 

The gene CTCF was duplicated in early amniotes, giving rise to a paralogue called "Brother of Regulator of Imprinted Sites" (BORIS, or CTCFL) [@Loukinov:2002vc; @Hore:2008wr]. Human CTCF and CTCFL have on average 99.5% identity within the ZF domain but much less conserved in N and C terminal domains. BORIS became progressively specialized to testis-specific expression throughout amniote evolution. CTCFL is normally expressed only in germ cells but also aberrantly activated in numerous cancers [@Pugacheva:2010tl][@Nishana:2020tv]. CTCFL and CTCF binding partly overlap, but given that its binding does not overlap with cohesin [@Sleutels:2012vd], it cannot insulate chromosome domains [@Nishana:2020tv]. CTCFL sites had less of a requirement for the "A" base in the triplet where ZF7 binds and an increase for "C" in the triplet that was bound by ZF4 [@Nishana:2020tv][@Sleutels:2012vd].


We utilized the PWMscan web tool [@Ambrosini:2018wv] because it has a large collection of PWMs (JASPAR, HOCOMOCO, SwissRegulon, UniPROBE, CIS-BP, data from Jomla, Isakova publications) and genomes (49 genome assemblies); its results show over 99% agreement with FIMO, HOMER, and other motif scanning tools.

# Methods



# Results



Chang et al. provides CTCF ChIP-seq peaks from mouse ESCs [@Chang:2021wl]. They report 174,268 peaks (83,522 unique regions) having at least one significant CTCF binding motif (mm10 genome assembly). It should be noted that mouse embryonic stem cells (ESCs) have thousands of CTCF-binding events that cannot be conserved in the human genome, as they are found in rodent- specific B2 repeat elements [@Bourque:2008ws].

CTCFBSDB assembled predicted and experimentally detected CTCF binding information [@Bao:2008vq]. Human predictions containes 13,356 sites and mouse predictions contained 5,504 sites (hg19 and mm9 genome assemblies, respectively). Unfortunately, the database has not been updated since 2012; therefore, its data is of limited use.

The ENCODE Phase III data release has created SCREEN, a registry of cis-regulatody element (cCRE) regions for human (hg38) and mouse (mm10) genome assemblies [@ENCODE-Project-Consortium:2020uq]. The human set of CTCF-bound cCREs contains 450,641 records. They were classified into distal-bound Enhancer-Like Signature (dELS) regions (61.18%), proximal ELS (pELS) regions (21.48%), CTCF-only sites (7.95%), promoter-like signature (PLS) regions (6.98%), and DNase-H3K4me3 regions bound by CTCF (1.84%). The mouse set contained 82,777 records similarly classified into 36.46% dELS-bound CTCF sites, 25.62% CTCF-only sites, 20.74% pELS-bound, 13.06 PLS-bound, and 4.11% DNase-H3K4me3-bound CTCF sites.

How to insert single [@Dozmorov:2016aa][@Glass:2016aa] or multiple references [@Dozmorov:2016aa; @Glass:2016aa].

+-------+
| ![](figures/chicken.png)      |
| **Figure 1. How to insert a figure.** Chicken chicken chicken.     |
+-------+
# Discussion

# Abbreviations

# Acknowledgements

_Conflict of Interest._ None.

# Funding

# Tables

# Figure legends

# Supplementary material

# References
