---
# csl: /Volumes/GoogleDrive/My Drive/manuscript_template/styles.ref/cell-reports.csl
csl: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/styles.ref/cell-reports.csl
output: 
  word_document:
#    reference_docx: /Volumes/GoogleDrive/My Drive/manuscript_template/styles.doc/NIH_grant_style.docx
    reference_docx: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/styles.doc/NIH_grant_style.docx
# https://github.com/mdozmorov/manuscript_template/tree/master/references
# bibliography: /Volumes/GoogleDrive/My Drive/manuscript_template/references/references_MD.bib
bibliography: /Users/mdozmorov/Documents/Work/presentations/Templates/manuscript_template/references/references_MD.bib
---

Authors

# Abstract

### Keywords

# Introduction

The structural and regulatory organization of the mammalian genome is fundamentally dependent on CTCF (CCCTC-binding factor), a versatile transctiprion regulator that is evolutionary conserved from fruit fly to human [@Phillips:2009aa][@Ohlsson:2001wu][@Kim:2007ux]. Originally defined as a repressor/insulator protein [@Lobanenkov:1990vm][@Filippova:1996tv], it has been found to be involved in a variety of regulatory functions including transcriptional activation, imprinting, X-chromosome activation, the intra- and inter-chromosomal contacts in three dimensions [@Phillips:2009aa][@Ohlsson:2010uy]. CTCF is ubiquitously expressed, and its deletion in the germline cells is incompatible with cell viability [@Heath:2008vs]. Point mutation and loss of heterozygosity of CTCF is associated with human cancer, identifying CTCF as an important candidate tumor-suppressor gene [@Filippova:2002uz].

Most of CTCF binding sites are located far from the transcriptional start sites, with their distribution strongly correlated with genes [@Kim:2007ux].


 with binding motif length varying between 20 and 40bp

CTCF binds to different DNA sequences through combinatorial use of its 11-zinc finger domains to target sites that have remarkable sequence variation [@Filippova:1996tv][@Ohlsson:2001wu][@Jolma:2013vd]. A 15-20bp CTCF consensus motif from the analysis of ChIP-seq data has been defined [@Kim:2007ux], referred hereafter as M1. This motif interacts with the N-terminal zinc fingers for the majority of CTCFâ€“DNA-binding events. Subsequently, a shorter motif (M2) [@Schmidt:2012uk], interacting with the C-terminal fingers. The complexity of CTCF binding was also noted in the earlier study of motif discovery in conserved noncoding elements by Xie et al. [@Xie:2007wm], detecting LM3, LM7, LM23 motifs as parts of CTCF binding. Subsequently, $\sim80%$ of CTCF conserved motifs were found to be bound by ZFs 4-7, while ZFs 1-2 and 8-11 bind to nonconserved flanking sequences and ZFs 9-11 associate with a second conserved motif [@Nakahashi:2013uw][@Pugacheva:2020wz]. Studies have reported 20,000-50,000 CTCF binding sites within the human and mouse genomes [@Kim:2007ux][@Nakahashi:2013uw][@Wang:2012ty].

CTCF binding has long been known to be critical for Topologically Associating Domain (TAD) and chromatin loop formation [@Dixon:2012aa][@Oti:2016aa][@Vietri-Rudan:2015aa]. Loops are formed via cohesin-mediated loop extrusion mechanism [@Alipour:2012aa][@Fudenberg:2016aa][@Goloborodko:2016vz][@Schwarzer:2016aa] with CTCF (in convergent orientation) and cohesin marking domain boundaries [@Nishana:2020tv][@Rao:2014aa][@Vietri-Rudan:2015tu][@Beagan:2020wz][@Stilianoudakis:2020aa][@Pugacheva:2020wz]. To demonstrate the use of CTCF data package, we provide a Python script that quantifies the proportion of loops 

The gene CTCF was duplicated in early amniotes, giving rise to a paralogue called "Brother of Regulator of Imprinted Sites" (BORIS, or CTCFL) [@Loukinov:2002vc; @Hore:2008wr]. Human CTCF and CTCFL have on average 99.5% identity within the ZF domain but much less conserved in N and C terminal domains. BORIS became progressively specialized to testis-specific expression throughout amniote evolution. CTCFL is normally expressed only in germ cells but also aberrantly activated in numerous cancers [@Pugacheva:2010tl][@Nishana:2020tv]. CTCFL and CTCF binding partly overlap, but given that its binding does not overlap with cohesin [@Sleutels:2012vd], it cannot insulate chromosome domains [@Nishana:2020tv]. CTCFL sites had less of a requirement for the "A" base in the triplet where ZF7 binds and an increase for "C" in the triplet that was bound by ZF4 [@Nishana:2020tv][@Sleutels:2012vd].

We aimed to collect general-purpose CTCF binding sites for human and mouse genomes, the top two species by the number of experiments in GEO. CTCF binding profiles between different tissues exhibit substantial concordance [@Kim:2007ux]. For instance, up to 70% of binding sites are common between any two given cell types, while variability appears to result from differential DNA methylation [@Wang:2012ty]. 

# Methods

MEME motif databases (updated 20 Mar 2022) containing HOCOMOCO, CIS-BP, JASPAR2022, Jolma 2013, and SwissRegulon motifs in MEME format were downloaded from https://meme-suite.org/meme/meme-software/Databases/motifs/motif_databases.12.23.tgz. Motifs from the CTCFBSDB database were downloaded from https://insulatordb.uthsc.edu/download/CTCFBSDB_PWM.mat and converted to MEME format using `transfac2meme -use_acc CTCFBSDB_PWM.mat > CTCFBSDB_PWM.meme`. Human and mouse motifs from CIS-BP were downloaded via the bulk download interface, combined and converted to the MEME format using a custom script. Motifs were visualized and clustered using the `motifStack` v.1.38.0 R package. Jolma 2013 and CTCFBSDB EMBL_M1 motifs were converted to their reverse complements to match orientation of other motifs (`universalmotif` v.1.24.4). 

CTCF binding sites were detected using the FIMO tool from the MEME v.4.11.2 suite [@Bailey:2009aa] was used for motif scanning using default parameters. Softmasked genome sequences were downloaded from the USCS Genome Browser database with the exception of T2T genome (chm13v2.0) downloaded from the https://github.com/marbl/CHM13 repository. Motifs were scanned separately for each chromosome (autosomal and sex chromosomes only), and the results were combined. We used human- and mouse-specific matrices for the corresponding genomes (HOCOMOCO, CIS-BP). As JASPAR, CTCFBSDB, and CIS-BP databases provide several CTCF motifs, all database-specific motifs were used to scan the genomes; consequently, the detected CTCF sites may overlap due to motif similarity. Predefined CTCF binding sites were obtained from the SCREEN ENCODE and CTCFBSDB databases. We did not include CTCFBSDB data compiled from public experimental data because 1) mouse data was lacking, 2) hg18 genome assembly limits usability, and 3) region size was highly variable, ranging from 2bp to >350Kb.

To investigate the effect of coordinate conversion, we used `liftOver` v.377 on CTCF binding sites (JASPAR 2022 MA0139.1) aligned to hg18, hg19, hg38, and T2T (NCBI GCA_009914755.4 assembly) genomes. The conversion chains were obtained from the UCSC Genome Browser database except for the T2T genome providing chains at the https://github.com/marbl/CHM13 repository. We performed all conversions except hg38->hg18 and hg18->T2T, T2T->hg18 due to lack of the corresponding chains. Jaccard overlap matrix between the original and liftOver CTCF binding sites was clustered using Euclidean distance and ward.D clustering method and visualized using the `pheatmap` v.1.0.12 R package.

# Results

CTCF binding specificities in a form of Position Weight Matrices (PWMs) [@Stormo:2000uw] are available from multiple databases. The JASPAR 2022 [@Castro-Mondragon:2022uv] database list one 19bp-long CTCF PWM (MA0139.1) and two 34 and 35bp-long PWMs (`Table`). The HOCOMOCO v11 [@Kulakovskiy:2018vw] database lists one CTCF PWM (H11MO.0.A) that is highly similar to the 19bp PWM in JASPAR (`Figure_clustered_main_PWMs`). Although human and mouse CTCF motifs are considered highly conserved [@Xie:2007wm], the HOCOMOCO database provides mouse-specific CTCF PWM which is expectedly similar to human (`Figure_clustered_main_PWMs`). SwissRegulon [@Pachkov:2013uk] provides one 20bp-long CTCF PWM similar to the JASPAR and HOCOMOCO PWMs. The 34bp and 35bp JASPAR PWMs formed their own cluster, together with the 17bp PWM identified in Jolma et al. study [@Jolma:2013vd] (`Figure_clustered_main_PWMs`). The CTCF binding site database (CTCFBSDB) provides multi-organism computationally predicted and experimentally determined from the literature CTCF binding sites [@Ziebarth:2013vj] and includes PWMs for the M1+M2 motifs identified by Schmidt et al. [@Schmidt:2012uk], the Ren_20 motif identified by Kim et al. [@Kim:2007ux], and the LM2, LM7, and LM23 motifs identified by Xie et al. [@Xie:2007wm] (`Figure_clustered_CTCFBSDB_PWMs.svg`). The Cis-BP database [@Weirauch:2014ul] lists 83 directly determined human-specific binding motifs (8 require Transfac license) ranging from 11 to 21bp (median - 15bp), likely reflecting the variability of CTCF motifs without considering the M1+M2 combinatorial structure (`Table`). These PWMs were used to scan human and mouse genomes for CTCF binding motifs.

Several databases and papers provide predicted and experimental CTCF binding information. The ENCODE Phase III data release has created SCREEN, a registry of cis-regulatody element (cCRE) regions for human (hg38) and mouse (mm10) genome assemblies [@ENCODE-Project-Consortium:2020uq]. The human set of CTCF-bound cCREs contains 450,641 records (median size 260bp), classified into distal-bound Enhancer-Like Signature (dELS) regions (61.18%), proximal ELS (pELS) regions (21.48%), CTCF-only sites (7.95%), promoter-like signature (PLS) regions (6.98%), and DNase-H3K4me3 regions bound by CTCF (1.84%). The mouse set contained 82,777 records (median size 284bp) similarly classified into 36.46% dELS-bound CTCF sites, 25.62% CTCF-only sites, 20.74% pELS-bound, 13.06 PLS-bound, and 4.11% DNase-H3K4me3-bound CTCF sites. A limitation of this data is that it lacks strand information. We added human and mouse CTCF-bound cCREs (hg38 and mm10 genome assemblies) as experimentally validated CTCF binding sites.

The CTCFBSDB database [@Bao:2008vq] assembled predicted CTCF binding information for human (13,401 20bp-long sites) and mouse (5,504 sites) genomes. The database has not been updated from 2013 and the predictions are available for hg18 and mm8 genome assemblies only (`Table`). Coordinate conversion between genome assemblies is a well-established practice; consequently, we investigated Jaccard overlap among CTCF sites aligned to hg18, hg19, hg38, and T2T genome assemblies and their liftover versions. Expectedly, genome assembly was the primary driver of similarity among sets of CTCF sites, irrespectively whether they were detected in the original genome assembly or lifted over from those detected in other assemblies (`Figure_liftOverJaccard.svg`). liftOver among hg18, hg19, and hg38 showed the highest Jaccard overlap (0.96-1.00), while CTCF sites lifted from hg19 and hg38 to T2T showed less overlap (0.92-0.93) due to the relatively large number of unmapped sites (`Table_liftOver_stats.xlsx`). These results indicate that liftOver is a viable alternative to obtain CTCF sites in the required genome assembly. We provide hg19 and hg38 versions of predicted CTCF sites from the CTCFBSDB database.


Chang et al. provides CTCF ChIP-seq peaks from mouse ESCs [@Chang:2021wl]. They report 174,268 peaks (83,522 unique regions) having at least one significant CTCF binding motif (mm10 genome assembly). It should be noted that mouse embryonic stem cells (ESCs) have thousands of CTCF-binding events that cannot be conserved in the human genome, as they are found in rodent- specific B2 repeat elements [@Bourque:2008ws].





Jolma et al. [@Jolma:2013vd] analyzed the sequence-specific binding of human TFs using high-throughput SELEX and ChIP sequencing. This study defined 830 binding profiles were obtained, describing 239 distinctly different binding specificities. They defined a 17bp long CTCF motif dubbed CTCF_full.


How to insert single [@Dozmorov:2016aa][@Glass:2016aa] or multiple references [@Dozmorov:2016aa; @Glass:2016aa].

+-------+
| ![](figures/chicken.png)      |
| **Figure 1. How to insert a figure.** Chicken chicken chicken.     |
+-------+

# Discussion

Our goal was to provide easy programmatic access to CTCF binding data that is generally applicable for any cell or tissue type. For those interested in cell-type-specific CTCF binding data, many databases, such as ENCODE, GEO, UniBind, ReMap, CistromeDB, provide download and/or REST API access to it. Bioconductor's AnnotationHub itself contains human and mouse CTCF binding data from ENCODE. Recent methods for cell-type-specific prediction of TFBSs, such as Catchitt, also provide downloadable data. Given the use of cell-type-specific data is specialized, we aimed to provide broadly applicable CTCF binding data.


Besides FIMO, other tools for motif scanning can be utilized. They have their own algorithms and defaults which may lead to different results. The popular HOMER motif screening tool [@Heinz:2010aa] has several options, including its unique autonormalization of sequencing bias algorithm. The PWMscan genomewide motif scanning web tool [@Ambrosini:2018wv] uses its own matrix_scan C program. PWMscan has the default background nucleotide composition A=0.29, C=0.21, G=0.21, T=0.29 instead of the FIMO's default A=0.25, C=0.25, G=0.25, T=0.25, and uses a 1E-5 p-value threshold instead of 1E-4 for FIMO. PWMscan performed extensive benchmarking of its scanning tool and showed over 99% agreement with FIMO, HOMER, and other motif scanning tools. Our data is includes results using FIMO tool and we expect these results to be generally consistent when using other tools.




# Abbreviations

# Acknowledgements

_Conflict of Interest._ None.

# Funding

# Tables

# Figure legends

# Supplementary material

# References
